scripts/shtf.py

```diff
@@ -0,0 +1,536 @@
+    notification = {}
+    for section_name in cfile.sections():
+
+        if "chains" in section_name:
+            chain_conf = cfile[section_name]
+            chain_id = int(chain_conf.get("chain-id"))
+
+            rpcs[chain_id] = (
+                chain_conf.get("rpc-url"),
+                BlockNumber(int(chain_conf.get("start-block"))),
+            )
+            explorers[chain_id] = chain_conf.get("explorer")
+
+        if "notification.rocket" in section_name:
+            notification_conf = cfile[section_name]
+
+            notification["system"] = "rocket"
+            notification["url"] = notification_conf.get("url")
+            notification["channel"] = notification_conf.get("channel")
+
+    global_config = {
+        "agent_address": agent_address,
+        "deployment_dir": deployment_dir,
+        "rpcs": rpcs,
+        "explorers": explorers,
+        "notification": notification,
+    }
+
+
+def get_config() -> ConfigType:
+    assert global_config is not None
+    return global_config
+
+
+@click.command
+@click.option(
+    "--config",
+    "-c",
+    type=click.File("r"),
+    required=True,
+    default="monitoring-mainnet.conf",
+    help="Config file with chains configuration",
+)
+def main(
+    config: Iterable[str],
+) -> None:
+    setup_logging(log_level="DEBUG", log_json=False)
+
+    set_config(config)
+
+    transfers = create_transfers_object(fetch_events())
+
+    cleanup_transfers(transfers)
+
+    ctx = Context()
+    ctx.agent_address = get_config()["agent_address"]
+
+    analyze_transfers(transfers, ctx)
+
+
+def cleanup_transfers(transfers: transfersType) -> None:
+    for requestId, values in transfers.copy().items():
+        if values.get("created") is None:
+            del transfers[requestId]
+
+        claims = values.get("claimed")
+        if claims is not None:
+            for events in claims.values():
+                events.sort(key=lambda a: -1 if a.last_challenger == ADDRESS_ZERO else 1)
+
+
+def fetch_events() -> ChainToEvent:
+    deployment_info = beamer.contracts.load_deployment_info(get_config()["deployment_dir"])
+    events = {}
+
+    for chain_id, (rpc, start_block) in get_config()["rpcs"].items():
+        web3 = make_web3(rpc)
+        assert chain_id == ChainId(web3.eth.chain_id)
+
+        info = deployment_info[ChainId(chain_id)]
+        contracts = beamer.contracts.make_contracts(web3, info)
+
+        ef = beamer.events.EventFetcher(
+            web3, (contracts["RequestManager"], contracts["FillManager"]), start_block
+        )
+
+        events[chain_id] = ef.fetch()
+
+    return cast(ChainToEvent, events)
+
+
+def make_web3(rpc: str) -> Web3:
+    web3 = Web3(Web3.HTTPProvider(rpc, request_kwargs=dict(timeout=5)))
+    web3.middleware_onion.inject(geth_poa_middleware, layer=0)
+    return web3
+
+
+def create_transfers_object(events_grouped_by_chain_id: ChainToEvent) -> transfersType:
+    transfers = defaultdict(dict)  # type: ignore[var-annotated]
```

Comment:
I would be happy if someone can let me know how to type transfers here as transfersType?

Reply:
`
from typing import cast

transfers = cast(transfersType, transfers)
`

Reply:
and normally new type should be also `TransfersType` instead of `transfersType`